test/request_source/request_source_plugin_test.cc

```diff
@@ -0,0 +1,179 @@
+#include "envoy/common/exception.h"
+
+#include "external/envoy/source/common/config/utility.h"
+#include "external/envoy/test/mocks/api/mocks.h"
+#include "external/envoy/test/mocks/stats/mocks.h"
+#include "external/envoy/test/test_common/file_system_for_test.h"
+#include "external/envoy/test/test_common/utility.h"
+
+#include "request_source/request_options_list_plugin_impl.h"
+
+#include "test/request_source/stub_plugin_impl.h"
+#include "test/test_common/environment.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Nighthawk {
+
+namespace {
+using nighthawk::request_source::FileBasedPluginConfig;
+using nighthawk::request_source::StubPluginConfig;
+using ::testing::NiceMock;
+using ::testing::Test;
+
+class StubRequestSourcePluginTest : public Test {
+public:
+  StubRequestSourcePluginTest() : api_(Envoy::Api::createApiForTest(stats_store_)) {}
+  Envoy::Stats::MockIsolatedStatsStore stats_store_;
+  Envoy::Api::ApiPtr api_;
+};
+
+class FileBasedRequestSourcePluginTest : public Test {
+public:
+  FileBasedRequestSourcePluginTest() : api_(Envoy::Api::createApiForTest(stats_store_)) {}
+  Envoy::Stats::MockIsolatedStatsStore stats_store_;
+  Envoy::Api::ApiPtr api_;
+  nighthawk::request_source::FileBasedPluginConfig
+  MakeFileBasedPluginConfigWithTestYaml(absl::string_view request_file) {
+    nighthawk::request_source::FileBasedPluginConfig config;
+    config.mutable_file_path()->assign(request_file);
+    config.mutable_max_file_size()->set_value(4000);
+    return config;
+  }
+};
+
+TEST_F(StubRequestSourcePluginTest, CreateEmptyConfigProtoCreatesCorrectType) {
+  auto& config_factory =
+      Envoy::Config::Utility::getAndCheckFactoryByName<RequestSourcePluginConfigFactory>(
+          "nighthawk.stub-request-source-plugin");
+  const Envoy::ProtobufTypes::MessagePtr empty_config = config_factory.createEmptyConfigProto();
+  const nighthawk::request_source::StubPluginConfig expected_config;
+  EXPECT_EQ(empty_config->DebugString(), expected_config.DebugString());
```

Comment:
We shouldn't compare protocol buffers using their string representation, since that isn't their API, just a debug string for humans to read. It isn't guaranteed to capture all the important details, nor it is guaranteed to be stable.

The gmock test suite comes with ::testing::EqualsProto helper we should be able to utilize in these cases.

Something like:

EXPECT_THAT(empty_config, ::testing::EqualsProto(expected_config))

Reply:
Please ignore this comment, looks like `::testing::EqualsProto` isn't as of yet available publically:

google/googletest#1761